cinemy/cinema/admin.py

```diff
@@ -38,10 +46,51 @@ class ReservationAdmin(NestedModelAdmin):
     exclude = ("is_reserved",)
 
 
+class CsvImportForm(forms.Form):
+    csv_file = forms.FileField()
+
+
 @admin.register(Movie)
 class MovieAdmin(NestedModelAdmin):
     inlines = (PlayingTimeInline,)
     exclude = ("imdb_link",)
+    change_list_template = "admin/movie_admin.html"
+
+    def get_urls(self):
+        urls = super().get_urls()
+        defined_urls = [
+            path("import_csv/", self.import_csv),
+        ]
+        return defined_urls + urls
+
+    def import_csv(self, request):
+        if request.method == "POST":
+            file = TextIOWrapper(
+                request.FILES["csv_file"].file, encoding=request.encoding
+            )
+            movies_from_csv = csv.DictReader(file)
+            movies = []
+            for movie_from_csv in movies_from_csv:
+                try:
+                    created_movie = Movie(**movie_from_csv)
+                    name = movie_from_csv["poster"].split("/")[-1]
+                    response = requests.get(created_movie.poster)
+                    if response.status_code == 200:
+                        created_movie.poster.save(
+                            name, ContentFile(response.content), save=False
+                        )
+                except requests.exceptions.MissingSchema:
```

Comment:
when you have 2 or more exceptions that are handled the same way, you can write them like this `except (requests.exceptions.MissingSchema, IntegrityError)` or, newly, in python 3.10 or 3.11 (not sure which one) you can do it without paranthesis.

But these look to me like they could use some handling. I mean, if you were to import some data in admin and the functionality fails silently, it would be a bit frustrating to not know what happened since the import didn't work and no error message is shown.

Reply:
Combined the two exceptions in one line and added messages in the UI in case the exception is caught.